.github/workflows/changeset.yml

```diff
@@ -0,0 +1,20 @@
+name: Changeset
+
+on:
+  pull_request:
+
+jobs:
+  changeset:
+    name: Validate changeset
+    runs-on: depot-ubuntu-22.04-16
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup
+        uses: ./.github/actions/setup
+
+      - name: Changeset
+        run: pnpm changeset status --since origin/main
```

Comment:
Have to do it like this due to base branch set as `main`. Relevant issue: https://github.com/changesets/changesets/issues/1055